### DISCUSSION TITLE
Display my cursor in the remote desktop.

### DISCUSSION THREAD

**Discussion by Globalrun** [rustdesk/rustdesk#4396] [Q&A] [answered]
When I watch somebody's desktop, it would be helpful, to use my cursor to guide the remote desktop.

**Comment by zaher** [comment]
I need this too, It is good to teaching other on remote computer

**Reply by rxsto** [reply]
I also want to emphasize how important this feature is, since explaining certain things just becomes impossible when there's no way of seeing the cursor of the remote user that is currently watching your screen. This is also a massive game changer that forces me and many of my colleagues back to AnyDesk.
Cheers

**Comment by Marcopato** [comment]
Hi everyone,

I hope you're all doing well. I wanted to suggest a feature that I believe would be incredibly useful for those of us using RustDesk for remote teaching and support. It would be great if RustDesk could implement a feature that allows both the instructor's and the student's cursors to be visible simultaneously on the student's screen. This would make it much easier to guide students without taking control of their mouse, using the cursor as a pointer instead.

This feature would be particularly beneficial for educational purposes, as it would enhance the interactivity and clarity of remote lessons. I believe many users would appreciate this addition.

Thank you for considering this suggestion!

Best regards, Marco A.F.

**Comment by ngotzhein** [comment]
In my opinion this is a very important feature. Could someone from the rustdesk team please consider adding this to the priority list. We need it for pair programming.

**Comment by danielabbatt** [comment]
Right - I believe this does exist, the reason I never found it is because usually I have disabled remote keyboard mouse. If you give control, the remote end then gets a keyboard icon appear at the top with the monitor selection etc. This has a "View mode" which I have ticked and just under that is an option to show the remote cursor, you can select that and then the host can turn off keyboard/mouse control again, the menu disappears on the client but the feature should then be working. I have found that the host needs to not be moving their mouse for the viewers mouse to show as moving after a second or two, even if in view mode. Ideally in view mode this should always be visible and moving. Also this doesn't seem to quite scale properly across multiple monitors though.

**Comment by FloMaetschke** [comment]
i am waiting for this teamviewer feature in rustdesk for multiple years now... still missing. but would be so powerful for pair progamming etc.

**Comment by AndyBodnar** [comment]
This would be super useful for remote support sessions! Currently RustDesk follows the standard remote desktop model where your cursor controls the remote cursor directly.

A few implementation ideas:

1. **Dual cursor mode** - Show both your local cursor and the remote cursor, with yours being a different color/style (like a laser pointer)

2. **Annotation overlay** - Similar to Zoom, let the viewer draw/point on screen without affecting the remote session

3. **Follow mode** - Your cursor movements are visible but dont control anything, just highlight where youre pointing

For now, some workarounds:
- Use the built-in annotation/drawing tools if available
- Share your screen back to the other person so they can see what youre pointing at
- Use keyboard shortcuts + verbal guidance

This is definitely a valid feature request - would make teaching and support much easier. Upvoting this!

**Comment by rustdesk** [comment, marked as answer]
Ready since 1.4.2: 
https://x.com/rustdesk/status/1960981998668829049
https://x.com/rustdesk/status/1964862556604088383
